### PR TITLE
Migrates android semanitcs integration to integration test

### LINKED ISSUE #124636: Linux_android android_semantics_integration_test is flaky on every commit
The flake happens since [0fa15e04acd8db185ab17e5d1d351ecbcbd7b51f](https://github.com/flutter/flutter/commit/0fa15e04acd8db185ab17e5d1d351ecbcbd7b51f)

Example build: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8784073433306279057/+/u/run_android_semantics_integration_test/test_stdout

```
[2023-04-11 15:26:46.173499] [STDOUT] stdout: [+1144 ms] 00:09 [32m+0[0m[31m -1[0m: AccessibilityBridge TextField TextField has correct Android semantics [1m[31m[E][0m[0m
[2023-04-11 15:26:46.173539] [STDOUT] stdout: [        ]   Expected: AndroidSemanticsNode with className: android.widget.EditText with actions: [AndroidSemanticsAction.click, AndroidSemanticsAction.copy, AndroidSemanticsAction.setSelection, AndroidSemanticsAction.setText] with flag isEditable: true with flag isFocusable: true with flag isFocused: true with flag isPassword: false
[2023-04-11 15:26:46.173550] [STDOUT] stdout: [        ]     Actual: AndroidSemanticsNode:<{flags: {isEditable: true, isEnabled: true, isDismissible: false, isCheckable: false, isChecked: false, isFocused: true, isLongClickable: false, isFocusable: true, isPassword: false}, id: 4, text: null, rect: {height: 144, width: 1080, bottom: 128.0, top: 80.0, left: 0.0, right: 360.0}, actions: [131072, 2097152, 16, 128], className: android.widget.EditText, liveRegion: 1, contentDescription: null}>
[2023-04-11 15:26:46.173559] [STDOUT] stdout: [        ]      Which: Expected actions: [AndroidSemanticsAction.click, AndroidSemanticsAction.copy, AndroidSemanticsAction.setSelection, AndroidSemanticsAction.setText]
[2023-04-11 15:26:46.173566] [STDOUT] stdout: [        ]             Actual actions: [AndroidSemanticsAction.clearAccessibilityFocus, AndroidSemanticsAction.click, AndroidSemanticsAction.setSelection, AndroidSemanticsAction.setText]
[2023-04-11 15:26:46.173575] [STDOUT] stdout: [        ]             Unexpected: {AndroidSemanticsAction.clearAccessibilityFocus}
[2023-04-11 15:26:46.173581] [STDOUT] stdout: [        ]             Missing: {AndroidSemanticsAction.copy}
[2023-04-11 15:26:46.180013] [STDOUT] stdout: [   +6 ms]   package:matcher/src/expect/expect.dart 149:31     fail
[2023-04-11 15:26:46.180055] [STDOUT] stdout: [        ]   package:matcher/src/expect/expect.dart 144:3      _expect
[2023-04-11 15:26:46.180063] [STDOUT] stdout: [        ]   package:matcher/src/expect/expect.dart 56:3       expect
[2023-04-11 15:26:46.180072] [STDOUT] stdout: [        ]   test_driver/main_test.dart 153:9                  main.<fn>.<fn>.<fn>
[2023-04-11 15:26:46.180079] [STDOUT] stdout: [        ]   ===== asynchronous gap ===========================
[2023-04-11 15:26:46.180088] [STDOUT] stdout: [        ]   package:test_api/src/backend/declarer.dart 215:9  Declarer.test.<fn>.<fn>
[2023-04-11 15:26:46.180095] [STDOUT] stdout: [        ]   ===== asynchronous gap ===========================
[2023-04-11 15:26:46.180102] [STDOUT] stdout: [        ]   package:test_api/src/backend/declarer.dart 213:7  Declarer.test.<fn>
[2023-04-11 15:26:46.180126] [STDOUT] stdout: [        ]   ===== asynchronous gap ===========================
[2023-04-11 15:26:46.180134] [STDOUT] stdout: [        ]   package:test_api/src/backend/invoker.dart 258:9   Invoker._waitForOutstandingCallbacks.<fn>
[2023-04-11 15:26:46.180156] [STDOUT] stdout: [        ] 00:09 [32m+0[0m[31m -1[0m: AccessibilityBridge TextField password TextField has correct Android semantics[0m
[2023-04-11 15:26:46.879596] [STDOUT] stdout: [ +698 ms] 00:10 [32m+0[0m[31m -2[0m: AccessibilityBridge TextField password TextField has correct Android semantics [1m[31m[E][0m[0m
[2023-04-11 15:26:46.879695] [STDOUT] stdout: [        ]   Expected: AndroidSemanticsNode with className: android.widget.EditText with actions: [AndroidSemanticsAction.click, AndroidSemanticsAction.copy, AndroidSemanticsAction.setSelection, AndroidSemanticsAction.setText] with flag isEditable: true with flag isFocusable: true with flag isFocused: true with flag isPassword: true
[2023-04-11 15:26:46.879728] [STDOUT] stdout: [        ]     Actual: AndroidSemanticsNode:<{flags: {isEditable: true, isEnabled: true, isDismissible: false, isCheckable: false, isChecked: false, isFocused: true, isLongClickable: false, isFocusable: true, isPassword: true}, id: 5, text: null, rect: {height: 144, width: 1080, bottom: 592.0, top: 544.0, left: 0.0, right: 360.0}, actions: [131072, 2097152, 16, 128], className: android.widget.EditText, liveRegion: 1, contentDescription: null}>
[2023-04-11 15:26:46.879755] [STDOUT] stdout: [        ]      Which: Expected actions: [AndroidSemanticsAction.click, AndroidSemanticsAction.copy, AndroidSemanticsAction.setSelection, AndroidSemanticsAction.setText]
[2023-04-11 15:26:46.879775] [STDOUT] stdout: [        ]             Actual actions: [AndroidSemanticsAction.clearAccessibilityFocus, AndroidSemanticsAction.click, AndroidSemanticsAction.setSelection, AndroidSemanticsAction.setText]
[2023-04-11 15:26:46.879799] [STDOUT] stdout: [        ]             Unexpected: {AndroidSemanticsAction.clearAccessibilityFocus}
[2023-04-11 15:26:46.879867] [STDOUT] stdout: [        ]             Missing: {AndroidSemanticsAction.copy}
[2023-04-11 15:26:46.880792] [STDOUT] stdout: [        ]   package:matcher/src/expect/expect.dart 149:31     fail
[2023-04-11 15:26:46.880859] [STDOUT] stdout: [        ]   package:matcher/src/expect/expect.dart 144:3      _expect
[2023-04-11 15:26:46.880895] [STDOUT] stdout: [        ]   package:matcher/src/expect/expect.dart 56:3       expect
[2023-04-11 15:26:46.880979] [STDOUT] stdout: [        ]   test_driver/main_test.dart 224:9                  main.<fn>.<fn>.<fn>
[2023-04-11 15:26:46.881011] [STDOUT] stdout: [        ]   ===== asynchronous gap ===========================
[2023-04-11 15:26:46.881031] [STDOUT] stdout: [        ]   package:test_api/src/backend/declarer.dart 215:9  Declarer.test.<fn>.<fn>
[2023-04-11 15:26:46.881090] [STDOUT] stdout: [        ]   ===== asynchronous gap ===========================
[2023-04-11 15:26:46.881158] [STDOUT] stdout: [        ]   package:test_api/src/backend/declarer.dart 213:7  Declarer.test.<fn>
[2023-04-11 15:26:46.881207] [STDOUT] stdout: [        ]   ===== asynchronous gap ===========================
[2023-04-11 15:26:46.881316] [STDOUT] stdout: [        ]   package:test_api/src/backend/invoker.dart 258:9   Invoker._waitForOutstandingCallbacks.<fn>
[2023-04-11 15:26:46.881486] [STDOUT] stdout: [        ] 00:10 [32m+0[0m[31m -2[0m: AccessibilityBridge TextField (tearDownAll)[0m
[2023-04-11 15:26:46.970480] [STDOUT] stdout: [  +88 ms] 00:10 [32m+0[0m[31m -2[0m: AccessibilityBridge SelectionControls (setUpAll)[0m
[2023-04-11 15:26:47.077444] [STDOUT] stdout: [ +106 ms] I/art     (12833): Do partial code cache collection, code=13KB, data=31KB
[2023-04-11 15:26:47.077524] [STDOUT] stdout: [        ] I/art     (12833): After code cache collection, code=13KB, data=31KB
```

Screenshot 2023-04-11 at 3 52 22 PM

### PR DESCRIPTION
**Pull request by chunhtai**
I think the flake is due to setclipboard or semantics update race condition. I migrated the test to use integration test package which relies less on timing

fixes https://github.com/flutter/flutter/issues/124636

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/integration_tests/android_semantics_testing/android/app/src/main/java/com/yourcompany/platforminteraction/MainActivity.java [hunk 1]
```diff
@@ -123,11 +122,60 @@ private Map<String, Object> convertSemantics(AccessibilityNodeInfo node, int id)
...
+                    actions.add(kIdByAction.get(action).intValue());
+                }
             }
             result.put("actions", actions);
         }
         return result;
     }
   }
+
+  // // These indices need to be in sync with android_semantics_testing/lib/src/constants.dart
```
Comment (goderbauer) [inline dev/integration_tests/android_semantics_testing/android/app/src/main/java/com/yourcompany/platforminteraction/MainActivity.java:135]:
```suggestion
  // These indices need to be in sync with android_semantics_testing/lib/src/constants.dart
```

File: dev/integration_tests/android_semantics_testing/test_driver/integration_test.dart [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import 'dart:io' as io;
```
Comment (goderbauer) [inline dev/integration_tests/android_semantics_testing/test_driver/integration_test.dart:1]: missing licence header?

File: dev/integration_tests/android_semantics_testing/lib/main.dart [hunk 3]
```diff
@@ -8,22 +8,19 @@ import 'dart:convert';
...
 void main() {
-  timeDilation = 0.05; // remove animations.
-  enableFlutterDriverExtension(handler: dataHandler);
   runApp(const TestApp());
 }
 
 const MethodChannel kSemanticsChannel = MethodChannel('semantics');
 
-Future<String> dataHandler(String? message) async {
+Future<String> invokeSemanticsMethod(String? message) async {
```
Comment (goderbauer) [inline dev/integration_tests/android_semantics_testing/lib/main.dart:23]: This method could probably just live in the test file now and be broken up in separate methods for getSemanticsNode and setClipboard? Also, the arguments can be passed directly, no need to turn them into a string `message`?

File: dev/integration_tests/android_semantics_testing/test_driver/main_test.dart [hunk 4]
```diff
@@ -1,702 +0,0 @@
...
-        int.parse(match.namedGroup('minor')!),
-        int.parse(match.namedGroup('patch')!),
-        build: match.namedGroup('build'),
-      );
-    }
-
-    setUpAll(() async {
-      driver = await FlutterDriver.connect();
-      talkbackVersion ??= await getTalkbackVersion();
-      print('TalkBack version is $talkbackVersion');
```
Comment (goderbauer) [inline dev/integration_tests/android_semantics_testing/test_driver/main_test.dart:82]: Looks like this no longer prints in the new test? Do we want to keep this? Could be useful for debugging...

File: dev/integration_tests/android_semantics_testing/test_driver/integration_test.dart [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  io.Process run = await io.Process.start(adbPath(), const <String>[
+    'shell',
+    'settings',
+    'put',
+    'secure',
+    'enabled_accessibility_services',
+    'com.google.android.marvin.talkback/com.google.android.marvin.talkback.TalkBackService',
+  ]);
+  await run.exitCode;
+  print('run.exitCode ${run.exitCode}');
```
Comment (goderbauer) [inline dev/integration_tests/android_semantics_testing/test_driver/integration_test.dart:26]: Is this print useful for anything?

File: dev/integration_tests/android_semantics_testing/test_driver/integration_test.dart [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+  await integrationDriver();
```
Comment (goderbauer) [inline dev/integration_tests/android_semantics_testing/test_driver/integration_test.dart:28]: Presumably, the test runs as part of this? Maybe leave a comment?

File: dev/integration_tests/android_semantics_testing/lib/main.dart (hunk 3 above)
Comment (goderbauer) [reply dev/integration_tests/android_semantics_testing/lib/main.dart:23]: Also, there shouldn't be a need to serialize the return value?

**Comment by chunhtai** [comment]
hmm, for some reason the `flutter test` would get permission denied when sending adb request, but `flutter drive` would not.

@christopherfujino Do you know why we may have different permission for running `flutter test` vs `flutter drive`?

This is the log
```
E/flutter (30629): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: ProcessException: Permission denied
E/flutter (30629):   Command: adb shell settings put secure enabled_accessibility_services com.google.android.marvin.talkback/com.google.android.marvin.talkback.TalkBackService
E/flutter (30629): #0      _ProcessImpl._start (dart:io-patch/process_patch.dart:401:33)
E/flutter (30629): #1      Process.start (dart:io-patch/process_patch.dart:38:20)
E/flutter (30629): #2      main (file:///Users/chtai/git/flutter/dev/integration_tests/android_semantics_testing/integration_test/main_test.dart:112:37)
E/flutter (30629): #3      _runMain.<anonymous closure> (dart:ui/hooks.dart:159:23)
E/flutter (30629): #4      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:296:19)
E/flutter (30629): #5      _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:189:12)
```

File: dev/integration_tests/android_semantics_testing/test_driver/integration_test.dart (hunk 5 above)
Comment (chunhtai) [reply dev/integration_tests/android_semantics_testing/test_driver/integration_test.dart]: removed

File: dev/integration_tests/android_semantics_testing/test_driver/main_test.dart (hunk 4 above)
Comment (chunhtai) [reply dev/integration_tests/android_semantics_testing/test_driver/main_test.dart:82]: added back to enableTalkBack

File: dev/integration_tests/android_semantics_testing/lib/main.dart (hunk 3 above)
Comment (chunhtai) [reply dev/integration_tests/android_semantics_testing/lib/main.dart]: refactored

File: dev/integration_tests/android_semantics_testing/test_driver/integration_test.dart (hunk 6 above)
Comment (chunhtai) [reply dev/integration_tests/android_semantics_testing/test_driver/integration_test.dart]: This file can be removed

**Comment by chunhtai** [comment]
I currently move the adb comment to enable talkback to devicelab test runner so I can workaround the issue that i can't call adb in `flutter test`

**Comment by chunhtai** [comment]
a friendly bump

File: dev/devicelab/lib/tasks/integration_tests.dart [hunk 7]
```diff
@@ -237,6 +241,10 @@ class IntegrationTest {
         ]);
       }
 
+      if (withTalkBack) {
+        await enableTalkBack();
```
Comment (goderbauer) [inline dev/devicelab/lib/tasks/integration_tests.dart:245]: Do we know here what device the test is targeting? Can we assert if someone accidentally sets `withTalkBack` to true on a device that doesn't support it (e.g. iOS)?

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**